theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -95,6 +95,7 @@ export class TraceViewerWidget extends ReactWidget {
                 if (this.isVisible) {
                     signalManager().fireTraceViewerTabActivatedSignal(experiment);
                 }
+                this.fetchMarkerSets(experiment.UUID);
```

Comment:
When you open a brand new trace from the Explorer with Open With > Trace Viewer, this instance is initialized with undefined traceUUID. So the marker sets are never updated.

I see they are updated when you later click the trace in the Opened Traces list, because a new instance is initialized. It actually opens a second tab for the same trace. Maybe that's another issue.